feature/security/gnsi/credentialz/tests/password_console_login/password_console_login_test.go

```diff
@@ -0,0 +1,310 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package password_console_login
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/openconfig/gnsi/credentialz"
+	tpb "github.com/openconfig/kne/proto/topo"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/binding"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	username          = "testuser"
+	password          = "i$V5^6IhD*tZ#eg1G@v3xdVZrQwj"
+	passwordVersion   = "v1.0"
+	passwordCreatedOn = 1705962293
+	// think this should work for all vendors contributing to fp
+	command = "show version"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+func createNativeRole(t testing.TB, dut *ondatra.DUTDevice, role string) {
+	t.Helper()
+	switch dut.Vendor() {
+	case ondatra.NOKIA:
+		roleData, err := json.Marshal([]any{
+			map[string]any{
+				"services": []string{"cli", "gnmi"},
+			},
+		})
+		if err != nil {
+			t.Fatalf("Error with json Marshal: %v", err)
+		}
+
+		userData, err := json.Marshal([]any{
+			map[string]any{
+				"role": []string{role},
+			},
+		})
+		if err != nil {
+			t.Fatalf("Error with json Marshal: %v", err)
+		}
+
+		SetRequest := &gpb.SetRequest{
```

Comment:
We are not setting username/password here but user-specific roles & assigning it to the testuser. Perhaps this can be refactored to simply assign admin role since the test is dealing with authentication & not authorization.

Reply:
I expect that we don't want vendor-specific role setting. The test should be the minimal bits possible to confirm authentication success. I agree though that both success and failure will have different messages - which is not helpful for testing. Ideally some expect type magic and uniformity of success/failure messages would get us to a vendor-neutral test?

Reply:
Removed vendor-specific role settings.